pkg/kubelet/dockertools/kube_docker_client.go

```diff
@@ -182,44 +182,40 @@ func (d *kubeDockerClient) InspectImage(image string) (*docker.Image, error) {
 		}
 		return nil, err
 	}
-	imageInfo := &docker.Image{}
+	imageInfo := &dockertypes.ImageInspect{}
 	if err := convertType(&resp, imageInfo); err != nil {
 		return nil, err
 	}
 	return imageInfo, nil
 }
 
-func (d *kubeDockerClient) ListImages(opts docker.ListImagesOptions) ([]docker.APIImages, error) {
-	resp, err := d.client.ImageList(getDefaultContext(), dockertypes.ImageListOptions{
-		MatchName: opts.Filter,
-		All:       opts.All,
-		Filters:   convertFilters(opts.Filters),
-	})
+func (d *kubeDockerClient) ListImages(opts dockertypes.ImageListOptions) ([]dockertypes.Image, error) {
+	images, err := d.client.ImageList(getDefaultContext(), opts)
 	if err != nil {
 		return nil, err
 	}
-	images := []docker.APIImages{}
-	if err = convertType(&resp, &images); err != nil {
-		return nil, err
+	dockerImages := []dockertypes.Image{}
+	for _, img := range images {
+		dockerImages = append(dockerImages, dockertypes.Image(img))
 	}
-	return images, nil
+	return dockerImages, nil
 }
 
-func base64EncodeAuth(auth docker.AuthConfiguration) (string, error) {
+func base64EncodeAuth(auth dockertypes.AuthConfig) (string, error) {
 	var buf bytes.Buffer
 	if err := json.NewEncoder(&buf).Encode(auth); err != nil {
 		return "", err
 	}
 	return base64.URLEncoding.EncodeToString(buf.Bytes()), nil
 }
 
-func (d *kubeDockerClient) PullImage(opts docker.PullImageOptions, auth docker.AuthConfiguration) error {
+func (d *kubeDockerClient) PullImage(opts dockertypes.ImagePullOptions, auth dockertypes.AuthConfig) error {
 	base64Auth, err := base64EncodeAuth(auth)
 	if err != nil {
 		return err
 	}
 	resp, err := d.client.ImagePull(getDefaultContext(), dockertypes.ImagePullOptions{
-		ImageID:      opts.Repository,
```

Comment:
No need to create a new `ImagePullOptions` now, just add `opts.RegistryAuth = base64Auth` is enough.